src/Agent.Listener/JobDispatcher.cs

```diff
@@ -48,42 +70,211 @@ public sealed class JobDispatcher : AgentService, IJobDispatcher
                             fileName: workerFileName,
                             arguments: "spawnclient " + pipeHandleOut + " " + pipeHandleIn,
                             environment: null,
-                            cancellationToken: token);
+                            cancellationToken: workerCancelTokenSource.Token);
```

Comment:
Consider renaming the variable or add a comment to be clear this is the token for the *process* only, whereas the token that's passed in is a higher level token for the worker... uh...